libraries/stdlib/genglsl/mx_normalmap.glsl

```diff
@@ -1,9 +1,9 @@
 void mx_normalmap_vector2(vec3 value, vec2 normal_scale, vec3 N, vec3 T, vec3 B, out vec3 result)
 {
-    value = (value == vec3(0.0)) ? vec3(0.0, 0.0, 1.0) : value * 2.0 - 1.0;
-
-    value = T * value.x * normal_scale.x + B * value.y * normal_scale.y + N * value.z;
-
+    value = (dot(value, value) == 0.0) ? vec3(0.0, 0.0, 1.0) : value * 2.0 - 1.0;
```

Comment:
Curious why you chose to use `dot(value, value)==0` here instead of just comparing each component.  Wouldn't the dot product here be more expensive?  And I think that comparing each component here would be clearer to understand. 

I also wonder if this might even be simplified even more to something like this... (or similar)
```
if (value.x != 0 || value.y != 0 || value.z != 0) {
  value =  value * 2.0 - 1.0;
  value = T * value.x * normal_scale.x +
            B * value.y * normal_scale.y +
            N * value.z;
    result = normalize(value);
} else {
  result = normalize(N);
}
```
This also nicely shows that if a zero valued input is passed we just return the normalized N.